### PR TITLE
Add cargo audit to github CI

### PR DESCRIPTION
**Pull request by ptulugu**

### CODE REVIEW

File: rita_exit/src/database/database_tools.rs [hunk 1]
```diff
@@ -637,15 +635,19 @@ pub fn get_client_subnet(
...
+    subprefix: u8,
 ) -> Result<IpNetwork, RitaExitError> {
-    // Convert IpNetwork into IPAdress type
-    let network: IPAddress = IPAddress::parse(exit_sub.to_string())
-        .expect("Paniced while parsing the exit subnet: IpNetwork -> IPAddress");
-    let mut net = network.network();
-    net.prefix = net.prefix.from(subprefix).unwrap();
+    let net;
+    let net_as_int: u128 = if let IpAddr::V6(addr) = exit_sub.network() {
+        net = Ipv6Network::new(addr, subprefix).unwrap();
```
Comment (jkilpatr) [inline rita_exit/src/database/database_tools.rs:642]: a comment for what this logic is doing would be good. I can't quite follow it.

File: rita_exit/src/database/database_tools.rs [hunk 2]
```diff
@@ -654,14 +656,19 @@ fn generate_iterative_client_subnet(
...
     // 2^(124 - 120) => 2^4 => 16
-    if ind < (1 << (subprefix - network.prefix.num)) {
-        net = net.from(&net.host_address, &net.prefix);
-        let size = net.size();
-        net.host_address += ind * size;
-        let ret = net
-            .to_string()
-            .parse()
-            .expect("Paniced while parsing exit subnet: IPAdress -> IpNetwork");
+    if ind < (1 << (subprefix - exit_sub.prefix())) {
```
Comment (jkilpatr) [inline rita_exit/src/database/database_tools.rs:659]: a comment about how this bitshifting works exactly would also be good.

**Review by jkilpatr — APPROVED** [review]
This looks great! I've put in a request for comments in a couple of spots.

File: rita_exit/src/database/database_tools.rs (hunk 1 above)
Comment (ptulugu) [reply rita_exit/src/database/database_tools.rs:642]: fixed

File: rita_exit/src/database/database_tools.rs (hunk 2 above)
Comment (ptulugu) [reply rita_exit/src/database/database_tools.rs:659]: there is already a comment above this

**Pull request merged**